rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/MultiChannelSchedulerDAO.java

```diff
@@ -60,52 +64,55 @@ int scheduleJob(
+            "    runningForChannel = tonumber(runningForChannel)\n" +
+            "  else \n" +
+            "    runningForChannel = 0\n" +
+            "  end\n" +
             "  if hasReady == 0 then\n" +
             //   as a result of RPOPLPUSH call above we know our channel is at the head of the list
             "    redis.call('LPOP', $multiChannelCircularBuffer$)\n" +
             "    redis.call('SREM', $multiChannelSet$, nextChannel)\n" +
             "  end\n" +
             "  local nextOffset = 0\n" +
-            "  while nextLimit > 0 and not isPaused and hasReady > 0 do\n" +
-            "     local jobs = redis.call('ZRANGEBYSCORE', readyQueue, 0, $now$, 'WITHSCORES', 'LIMIT', nextOffset, nextLimit)\n" +
-            "     if next(jobs) == nil then\n" +
-            "       break\n" +
-            "     end\n" +
-            "     for i=1,#jobs,2 do\n" +
-            "        local inRunningQueue = redis.call('ZSCORE', $runningQueue$, jobs[i])\n" +
-            "        if not inRunningQueue then\n" +
-            "            if perChannelTracking and perChannelLimit > 0 then\n" +
-            "              local runningForChannel = redis.call('GET', runningCount)\n" +
-            "              if runningForChannel and tonumber(runningForChannel) >= perChannelLimit then\n" +
-            "                break\n" +
-            "              end\n" +
-            "            end\n" +
-            "            reserved[reservedIndex] = jobs[i]\n" +
-            "            reserved[reservedIndex + 1] = jobs[i + 1]\n" +
-            "            redis.call('ZREM', readyQueue, reserved[reservedIndex])\n" +
-            "            redis.call('ZADD', $runningQueue$, $ttl$, reserved[reservedIndex])\n" +
-            "            if perChannelTracking then \n" +
-            "              redis.call('INCR', runningCount)\n" +
-            "            end\n" +
-            "            reservedIndex = reservedIndex + 2\n" +
-            "            nextLimit = nextLimit - 1\n" +
-            "            local hasReady = redis.call('ZCARD', readyQueue)\n" +
-            "            if hasReady == 0 then\n" +
-            //             as a result of RPOPLPUSH call above we know our channel is at the head of the list
-            "              redis.call('LPOP', $multiChannelCircularBuffer$)\n" +
-            "              redis.call('SREM', $multiChannelSet$, nextChannel)\n" +
-            "            end\n" +
+            "  while nextLimit > 0 and not isPaused and hasReady > 0 and not perChannelLimitReached(runningForChannel) do\n" +
+            "    local jobs = redis.call('ZRANGEBYSCORE', readyQueue, 0, $now$, 'WITHSCORES', 'LIMIT', nextOffset, nextLimit)\n" +
+            "    if next(jobs) == nil then\n" +
+            "      break\n" +
+            "    end\n" +
+            "    for i=1,#jobs,2 do\n" +
+            "      if perChannelLimitReached(runningForChannel) then\n" +
```

Comment:
but also check here in case we have filled up to our limit since starting. 

most of the rest of the changes here are formatting